### PR TITLE
setup-database-phase-2

### PR DESCRIPTION
**Pull request by MarhaensalenindoK**
https://www.figma.com/file/ORfEnDlqV63R9YzIogsiZ2/Poliklinik?node-id=921%3A30

### CODE REVIEW

File: app/Models/MedicalHistory.php [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        'update_at',
+    ];
+
+    public function patient() {
+        return $this->hasone(User::class, 'id', 'patient_id');
+    }
+
+    public function examiner() {
+        return $this->hasone(User::class, 'id', 'examiner_id');
+    }
```
Comment (altf4m88) [inline app/Models/MedicalHistory.php]: hasOne

File: app/Models/Queue.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        'update_at',
+    ];
+
+    public function patient() {
+        return $this->hasone(User::class, 'id', 'patient_id');
+    }
+
+    public function medicalHistory() {
+        return $this->hasone(MedicalHistory::class, 'id', 'medical_history_id');
+    }
```
Comment (altf4m88) [inline app/Models/Queue.php:35]: hasOne

File: app/Service/Database/ActionService.php [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        $query->where('medical_history_id', $medical_history_id);
+        $query->with('medicine');
+        $action = $query->paginate($per_page, ['*'], 'page', $page);
+
+        return $action->toArray();
+    }
+
+    public function detail($actionId)
+    {
+        $action = Action::with('medicine')->with('medicalHistory')->findOrFail($actionId);
```
Comment (altf4m88) [inline app/Service/Database/ActionService.php]:
```suggestion
        $action = Action::with(['medicine', 'medicalHistory'])->findOrFail($actionId);
```

File: app/Service/Database/QueueService.php [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        foreach ($attributes as $key => $value) {
+            $queue->$key = $value;
+        }
+
+        Validator::make($queue->toArray(), [
+            'medical_history_id' => 'required',
+            'patient_id' => 'required',
+            'queue' => 'required',
+            'date' => 'required',
+            'status' => 'required',
```
Comment (altf4m88) [inline app/Service/Database/QueueService.php]:
```suggestion
            'status' => ['required', Rule::in(config('constant.queue.status'))],
```

**Comment by altf4m88** [comment]
LGTM

**Comment by MarhaensalenindoK** [comment]
LGTM

**Pull request merged**